xborders

```diff
@@ -199,85 +177,120 @@ class Highlight(Gtk.Window):
-
-    notified_compositor = False
-    rect_x = None
-    rect_y = None
-    def _on_draw(self, _wid, ctx):
-        if not screen.is_composited():
-            self.move(1e6, 1e6) # move it off screen
-            if (self.notified_compositor == True):
-                return;
-            else:
-                subprocess.Popen(
-                    ["notify-send", "xborder", "xborders requires a compositor. Resuming once a compositor is found."]
-                )
-                self.notified_compositor = True;
+        self.set_keep_above(True)
+        self.set_title("xborders")
+        self.show_all()
+        self.border_path = [0, 0, 0, 0]
+
+        # Event connection:
+        self.connect("draw", self._draw)
+        self.connect("destroy", Gtk.main_quit)
+        self.connect('composited-changed', self._composited_changed_event)
+        self.wnck_screen.connect("active-window-changed", self._active_window_changed_event)
+
+    # This triggers every time the window composited state changes.
+    # https://docs.gtk.org/gtk3/signal.Widget.composited-changed.html
+    def _composited_changed_event(self, arg):
+        if self.screen.is_composited():
+            self.move(0, 0)
         else:
-            if (self.notified_compositor == True):
-                self.move(0, 0);
-            self.notified_compositor = False;
-        ctx.save()
-
-        if (not is_fullscreen(self.wnck_screen)):
-            win_data = get_win(self.wnck_screen)
-            if win_data != None:
-                x, y, w, h = win_data[0][0], win_data[0][1], win_data[1][0], win_data[1][1]
-                if BORDER_MODE == 0:
-                    x += BORDER_WIDTH / 2;
-                    y += BORDER_WIDTH / 2;
-                    w -= BORDER_WIDTH;
-                    h -= BORDER_WIDTH;
-                elif BORDER_MODE == 1:
-                    x -= BORDER_WIDTH / 2;
-                    y -= BORDER_WIDTH / 2;
-                    w += BORDER_WIDTH;
-                    h += BORDER_WIDTH;
-                draw.draw_rectangle(
-                    ctx,
-                    x,
-                    y,
-                    w,
-                    h,
-                    BORDER_RADIUS,
-                    BORDER_WIDTH,
-                    [0, 0, 0, 0],
-                    [BORDER_R, BORDER_G, BORDER_B, BORDER_A],
-                )
+            self.move(1e6, 1e6)
+            subprocess.Popen(["notify-send", "xborder", "xborders requires a compositor. Resuming once a compositor is running."])
+
+    # This event will trigger every active window change, it will queue a
+    # border to be drawn and then do nothing.
+    # See:
+    #    Signals available for the Wnck.Screen class: https://lazka.github.io/pgi-docs/Wnck-3.0/classes/Screen.html#signals
+    #    Signals available for the Wnck.Window class: https://lazka.github.io/pgi-docs/Wnck-3.0/classes/Window.html#signals
+    def _active_window_changed_event(self, screen, previous_active_window):
```

Comment:
Can you mark `screen` and `previous_active_window` as unused? also `arg` in `_composited_changed_event`

Reply:
It was an oversight, fortunately a ``"self.emit('composited-changed")``` in the constructor solves the black screen on start problem.

I did not test on gnome, but you are right, I'm relying on the ```geometry-changed``` event, since, when a window goes fullscreen, it changes geometry. This, as you pointed out, should not be the case. Fortunately the ```Wnck.Window``` class has a ```state-changed``` signal which I could use to setup a proper fullscreen event. 

I will mark those variables as unused.

I will setup a virtual machine to test another DEs, I'm genuinely curious to see this running under gnome.

Also, I'm working in a new commit, because I noticed this on stacking mode and I cannot unsee;
![image](https://user-images.githubusercontent.com/32084752/183234405-3de3b496-ebb4-4d49-bcf1-6552cb1066c8.png)

The program is now capable of drawing incomplete borders to avoid those situations, but it is incredibly buggy right now, I can setup a testing branch if you want to see it.
![image](https://user-images.githubusercontent.com/32084752/183234487-50404063-98ad-4f3e-b703-bebe32d9e063.png)

Reply:
Yes this should be in a separate PR.